src/react/11-state.md

```diff
@@ -0,0 +1,270 @@
+@page learn-react/manage-state Managing State
+@parent learn-react 11
+
+@description Manage component and application state with hooks
+
+@body
+
+## What is state? ##
+
+In React, `state` is any data determining the presention of your components. The value of a text box, whether or not a modal is open, and what page the app is currently showing are all potential examples of state.
+
+Using functional components, the `useState()` hook provides a value and setter capable of holding any native data type. This allows us to hold simple variables such as:
+
+```
+const [ count, setCount ] = useState(0)
+const [ flag, setFlag ] = useState(true)
+```
+
+And even hold more complex data.
+
+```
+const [ viewModel, setViewModel ] = useState(
+    {
+        showModal: false,
+        currentUser: {
+            name: 'Mike Myers'
+            id: 42
+        },
+        highlightItems: [1, 12, 15, 34, 66]
+    }
+)
+```
+
+**Important:** When using objects or arrays, you must provide a new merged state when using the setter. `useState` is immutable and will always replace any internal state.
+
+*For class based components see [this.setState()](https://reactjs.org/docs/state-and-lifecycle.html)*
+
+## Scope of State ##
+
+`useState` is scoped to the component it is declared in. For example, this simple component stores a simple string `displayText` and diplays its value as the user makes changes to a text input.
+
+```
+import React, { useState } from 'react'
+
+function DisplayComponent() {
```

Comment:
`DisplayComponent` is functionally ambiguous. Try `DisplayText` or `EditText`. I'd also suggest referring to it by name in lieu of "this simple component".

Speaking of which, right now this text uses "simple" too much in my opinion. It's very often unnecessary. I feel as though calling something simple may inadvertently make a reader feel incompetent should they disagree with your assessment. 

I like to use [The Hemmingway Editor](http://www.hemingwayapp.com/) to help me catch these issues. I recommend you run your text through it.